presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/LimitPushDown.java

```diff
@@ -139,7 +139,7 @@ public PlanNode visitAggregation(AggregationNode node, RewriteContext<LimitConte
                     node.getOutputSymbols().size() == node.getGroupingKeys().size() &&
                     node.getOutputSymbols().containsAll(node.getGroupingKeys())) {
                 PlanNode rewrittenSource = context.rewrite(node.getSource());
-                return new DistinctLimitNode(idAllocator.getNextId(), rewrittenSource, limit.getCount(), false, Optional.empty());
+                return new DistinctLimitNode(idAllocator.getNextId(), rewrittenSource, limit.getCount(), false, rewrittenSource.getOutputSymbols(), Optional.empty());
```

Comment:
I think it's saved by https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java#L463 and https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java#L396 and due to correct optimizers order (pruner runs first)